packages/flutter/lib/src/widgets/animated_scroll_view.dart

```diff
@@ -12,51 +12,164 @@ import 'scroll_view.dart';
+/// remove items. To refer to the [AnimatedListState] either provide a
+/// [GlobalKey] or use the static [of] method from an item's input callback.
 ///
-/// See also:
+/// This widget is similar to one created by [ListView.builder].
 ///
-/// * [AnimatedRemovedItemBuilder], a builder that is for removing items with
-///   animations instead of adding them.
-typedef AnimatedItemBuilder = Widget Function(BuildContext context, int index, Animation<double> animation);
-
-/// Signature for the builder callback used by widgets like [AnimatedGrid] (in
-/// [AnimatedGridState.removeItem]) to animated their children after they have
-/// been removed.
+/// {@youtube 560 315 https://www.youtube.com/watch?v=ZtfItHwFlZ8}
 ///
-/// The `context` argument is the build context where the widget will be
-/// created, and the `animation` is an [Animation] that should be used to
-/// animate an exit transition for the widget that is built.
+/// {@tool dartpad}
+/// This sample application uses an [AnimatedList] to create an effect when
+/// items are removed or added to the list.
+///
+/// ** See code in examples/api/lib/widgets/animated_list/animated_list.0.dart **
+/// {@end-tool}
 ///
 /// See also:
 ///
-/// * [AnimatedItemBuilder], a builder that is for adding items with animations
-///   instead of removing them.
-typedef AnimatedRemovedItemBuilder = Widget Function(BuildContext context, Animation<double> animation);
+///  * [SliverAnimatedList], a sliver that animates items when they are inserted
+///    or removed from a list.
+///  * [SliverAnimatedGrid], a sliver which animates items when they are
+///    inserted or removed from a grid.
+///  * [AnimatedGrid], a non-sliver scrolling container that animates items when
+///    they are inserted or removed in a grid.
+class AnimatedList extends _AnimatedScrollView {
+  /// Creates a scrolling container that animates items when they are inserted
+  /// or removed.
+  const AnimatedList({
+    super.key,
+    required super.itemBuilder,
+    super.initialItemCount = 0,
+    super.scrollDirection = Axis.vertical,
+    super.reverse = false,
+    super.controller,
+    super.primary,
+    super.physics,
+    super.shrinkWrap = false,
+    super.padding,
+    super.clipBehavior = Clip.hardEdge,
+  }) : assert(itemBuilder != null),
+       assert(initialItemCount != null && initialItemCount >= 0);
 
-// The default insert/remove animation duration.
-const Duration _kDuration = Duration(milliseconds: 300);
+  /// The state from the closest instance of this class that encloses the given
+  /// context.
+  ///
+  /// This method is typically used by [AnimatedList] item widgets that insert
+  /// or remove items in response to user input.
+  ///
+  /// If no [AnimatedList] surrounds the context given, then this function will
+  /// assert in debug mode and throw an exception in release mode.
+  ///
+  /// This method can be expensive (it walks the element tree).
```

Comment:
Should this method (and `maybeOf` below, and in other classes in the file) also mention that it doesn't create a dependency relationship?  Many/most of the `of` functions do that, so it might help clear up that they won't be getting rebuilt automatically if the state changes.